### PR TITLE
with embedded and corrections

### PR DESCRIPTION
**Pull request by Abibibi**

### CODE REVIEW

**Comment by ntchou** [comment]
When you click on quit button iframe doesn't close

image

same on firefox and chrome

**Comment by HugoGresse** [comment]
we cannot go back to the precedent url after creating a visio, it's maybe linked to react router and the history or how the new url is push to the browser

File: src/pages/Home/index.js [hunk 1]
```diff
@@ -1,13 +1,12 @@
 import React, { useState, useRef } from 'react'
 import styled from 'styled-components'
-import {Link} from 'react-router-dom'
+import {Link, Redirect, Route} from 'react-router-dom'
```
Comment (ntchou) [inline src/pages/Home/index.js:3]: `Redirect` is useless

File: src/pages/VideoCall/index.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+            daily.leave()
+        }
+
+        window.addEventListener('beforeunload', leavingCallPage)
+        // ComponentWillUnmount
+        return () => {
+            window.removeEventListener('beforeunload', leavingCallPage)
+        }
+       
+    }, [])
```
Comment (ntchou) [inline src/pages/VideoCall/index.js:81]: better

```suggestion
    }, [url, leftCallFrame])
```

File: src/pages/VideoCall/index.js [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+            <IframeStyled>
+                {
+                    !leftCallFrame && <iframe
+                        title="video call iframe"
+                        ref={videoFrame}
+                        allow="microphone; camera; autoplay"
+                    />
+                }
+                {
+                    leftCallFrame && <div>Vous avez bien quitté l'appel. Vous pouvez fermer cette page.</div>
```
Comment (ntchou) [inline src/pages/VideoCall/index.js:97]: missing translation ;)

File: src/pages/VideoCall/index.js [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+                }
+            </IframeStyled>
+            {!leftCallFrame && <LeaveButton onClick={leavingCallFrame}>
+                <p>Quitter l'appel</p>
```
Comment (ntchou) [inline src/pages/VideoCall/index.js:101]: missing translation ;)

File: src/pages/VideoCall/index.js [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+    let { videoName } = useParams()
+
+    const url = `https://instantvisio.daily.co/${videoName}`
+
+    const videoFrame = useRef(null)
+
+    // to display confirmation message 
+    // when user attempts leaving page
+    const leavingCallPage = (event) => {
+        /* // Cancel the event as stated by the standard.
```
[the remark this answers is not in the document]
Comment (Abibibi) [reply src/pages/VideoCall/index.js]: done

File: src/pages/VideoCall/index.js (hunk 2 above)
Comment (Abibibi) [reply src/pages/VideoCall/index.js]: added url.

had previously added "leftCallFrame" but the attached error is generated

leftCallFrame_error

Comment (ntchou) [reply src/pages/VideoCall/index.js]: very weird, i try with these ref and i didn't have the error

File: src/pages/Home/index.js [hunk 6]
```diff
@@ -47,12 +46,12 @@ export default function Home(){
             <Form onSubmit={submit} isSending={loading} errorSending={error ? true : false} />
             <div ref={formSubmissionMessage}>
                 {videoCallId &&
-                            <Route  
-                                render={() => {
-                                    window.location.href = `https://instantvisio.daily.co/${videoCallId}`
-                                    return null
-                                }} 
-                            />
+                    <Route
```
Comment (HugoGresse) [inline src/pages/Home/index.js:49]: why not using a redirect here?

**Pull request merged**